.dockerignore

```diff
@@ -1,10 +1 @@
-.git
-.gitignore
-etc/initdev/.vagrant
-src/cmd/pfs/pfs-roler
-src/cmd/pfs/pfs-volume-driver
-src/cmd/pfs/pfs
-src/cmd/pfsd/pfsd
-src/cmd/pps/pps
-src/cmd/ppsd/ppsd
-generate
+*
```

Comment:
Why are we cleaning out our `.dockerignore` here? Although most of this is no longer needed I think `.git` would still be good to exclude.

Reply:
We're not cleaning it out so much as specifying everything as something that docker should ignore. Since the code is mounted in as a host volume, we no longer want to upload anything to the build context. (This nets us at least a handful of seconds ... and in cases where your local directory gets cluttered, can save a fair bit of time).

Reply:
Makes sense. We may want to consider moving the Dockerfiles out of the top level directory now. We originally put them there so that they could add `src` but now that they're ignoring it there's no reason for them to be there. No need to do it as part of this review.

Reply:
Noted. Issue here: https://github.com/pachyderm/pachyderm/issues/2669